### PR TITLE
Rivers worldsynth

### PR DESCRIPTION
**Pull request by ftomassetti**
We already calculated rivermap and lakemap but then we used an ad-hoc routine to draw rivers, with questionable results.

This patch is built on top of #167 

See images in #119

### CODE REVIEW

File: worldengine/drawing_functions.py [hunk 1]
```diff
@@ -40,50 +40,16 @@ def draw_rivers_on_image(world, target, factor=1):
...
-        if world.ocean[pos[1], pos[0]]:
-            return
-        x, y = pos
-        for dx in range(factor):
-            for dy in range(factor):
-                target.set_pixel(x * factor + dx, y * factor + dy,
-                                 (0, 0, 128, 255))
-        _draw_river(world, target, _lowest_neighbour(world, pos), factor)
-
-    n_rivers = int(math.sqrt(world.width * world.height))
```
Comment (tcld) [inline worldengine/drawing_functions.py]: Now the math module doesn't need to be imported anymore.

Comment (ftomassetti) [inline worldengine/drawing_functions.py]: Removed, thanks

File: worldengine/drawing_functions.py [hunk 2]
```diff
@@ -40,50 +40,16 @@ def draw_rivers_on_image(world, target, factor=1):
...
-    for i in range(1, n_rivers):
-        candidates = []
-        for j in range(1, 10):
-            candidates.append(_pseudo_random_land_pos(world, i * j + j))
```
Comment (tcld) [inline worldengine/drawing_functions.py]: I think that _pseudo_random_land_pos() is now not used anymore, either. I would like to see it removed.^^

File: worldengine/drawing_functions.py [hunk 3]
```diff
@@ -40,50 +40,16 @@ def draw_rivers_on_image(world, target, factor=1):
...
+    for y in range(world.height):
+        for x in range(world.width):
+            if world.is_land((x, y)) and (world.river_map[x, y] > 0.0):
+                for dx in range(factor):
+                    for dy in range(factor):
+                        target.set_pixel(x * factor + dx, y * factor + dy, (0, 0, 128, 255))
+            if world.is_land((x, y)) and (world.lake_map[x, y] != 0):
+                for dx in range(factor):
+                    for dy in range(factor):
+                        target.set_pixel(x * factor + dx, y * factor + dy, (0, 100, 128, 255))
```
Comment (tcld) [inline worldengine/drawing_functions.py]: Simple and effective code, I don't see anything that could go wrong. Certainly a huge improvement over the kind of complicated previous code. (And obviously the results are a lot better, too. :) )

File: worldengine/drawing_functions.py (hunk 2 above)
Comment (ftomassetti) [inline worldengine/drawing_functions.py]: correct, removing

**Comment by tcld** [comment]
This will improve the results of world generation *so much*!

The code reminds me a little bit of what ```irrigation.py``` looked like before I replaced its core with numpy code - I guess that drawing the rivers and lakes is kind of a slow process? I wouldn't know how to improve it, I'm just curious.

**Comment by psi29a** [comment]
*thumbs up* just need new blessed images?

**Comment by ftomassetti** [comment]
it seems so. I will commit the new blessed images and merge then.

**Comment by ftomassetti** [comment]
Blessed images updated, tests restarted

**Comment by ftomassetti** [comment]
rebasing on master and then restarting tests

**Pull request merged**